infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -204,10 +204,25 @@ public synchronized void delete(@Nonnull String key,
 
     @Override
     public synchronized <T> T get(Class<T> tclass, String prefix, String key) {
-        String path = getKey(prefix, key);
+        return get(new KvRecord<>(prefix, key, tclass));
+    }
+
+    @Override
+    public synchronized <T> void delete(Class<T> tclass, String prefix, String key) {
```

Comment:
I'm not sure these methods can carry their weights and they add complexity.

Reply:
They are used in many places (actually everywhere), I created new methods (which use KvRecord) and made these methods - deprecated.